### DISCUSSION TITLE
Recommended workflow for mission specific GDS edits

### DISCUSSION THREAD

**Discussion by ebenezerarunkumar** [nasa/fprime#1496] [Q&A] [answered]
Hi! 

I'm working on a project where we are using the CCSDS framing/deframing protocol instead of the standard fprime protocol. In order to get the GDS working with a deployment using the CCSDS protocol, I have to make changes to the GDS source code, namely framing.py. 

Is there a way I can point the GDS to using a new framing.py file without having to fork the GDS repo and put it in my project repo?

Do you have any thoughts on this? Am I approaching this problem the correct way or am I misunderstanding how the GDS works?

**Comment by SterlingPeet** [comment]
@LeStarch I have been using [pluggy](https://pluggy.readthedocs.io/en/stable/) for the purpose of making pluggable/extensible python applications lately.  It seems to be able to solve the myriad of discovered requirements I uncovered while rolling my own library for this purpose.  Extra bonus, it is the system that ``pytest`` runs on, so its very well vetted and maintained.

Perhaps you would like to take a look?  It would likely be what is needed to really get this functionality off the ground.

**Comment by Joshua-Anderson** [comment, marked as answer]
I think you're approaching this the right way. Right now the GDS isn't designed to be easily extensible so if you want to modify portions like the framing you would need to clone the GDS and modify it yourself.
